### ISSUE TITLE
PluginBrowser: Tree layout and search bar

### ISSUE THREAD

**Issue by JohannesLorenz** [LMMS/lmms#4759]
This is a begginer task to prepare our plugin browser for LV2 (and the new layout). The following proposals are conforming to @budislav 's design and also favored by multiple LMMS devs. They were discussed in #4733 .

# Current look

```
<filter search bar>
bit invader
triple oscillator
zynaddsubfx
...
```

# Changes

* It should have a new search bar (like `EffectSelectDialog`)
* The `PluginDescWidget`s should be in a tree layout.

Aside from that, everything should stay like it is, i.e. keep the plugin names and the logos.

You should only create two top level items by now: LMMS and LV2. Move all current plugins to LMMS, leave LV2 empty.

```
<filter search bar>
LMMS
|- bit invader
|- triple oscillator
|- zynaddsubfx
|- ...
LV2
```

# Where to get started?

## Code

```
include/PluginBrowser.h
src/gui/PluginBrowser.cpp
```

## Links

E.g. (but you may find better ones):
* [How to add QPushButton or widgets to a QTreeWidget as QTreeWidgetItem](https://blog.manash.me/quick-qt-6-how-to-add-qpushbutton-or-widgets-to-a-qtreewidget-as-qtreewidgetitem-2ae9f54c0e5f?gi=e708ac9fed27)
* [QTreeViewFilter](https://www.qtcentre.org/threads/46471-QTreeView-Filter)

## Git

Open a new branch, push to your remote and make a pull request.

**Comment by itlives** [comment]
is this still open? I could try to work on the ticket

**Comment by JohannesLorenz** [comment]
@itlives As no one else asked to work on it, I'd consider it still open. Your work on this issue would be appreciated.

**Comment by itlives** [comment]
@JohannesLorenz i will try my best, still pretty new to the whole thing but ill take it as a challenge. Will get help if needed

**Comment by itlives** [comment]
@JohannesLorenz you might want to edit the suggested filepath to src/gui/PluginBrowser.cpp ;)

**Comment by itlives** [comment]
@JohannesLorenz i am now working on the issue. Should the search bar act like a filter or a sorter? (only display matching plugins OR show matching plugins on top)

**Comment by JohannesLorenz** [comment]
@itlives I'd prefer a filter, that would also be conforming to what we have in `EffectSelectDialog` (when you add an effect somewhere and are asked to choose one).

Though that's just my opinion. @tresf What would you say?

**Comment by tresf** [comment]
> > Should the search bar act like a filter or a sorter? (only display matching plugins OR show matching plugins on top)

> @itlives I'd prefer a filter[...] @tresf What would you say?

Agreed.

**Comment by JohannesLorenz** [comment]
@itlives How is it going? If there is anything you need help with, just let us know.

**Comment by itlives** [comment]
> @itlives How is it going? If there is anything you need help with, just let us know.

Hello, 
I am sorry to say that  I am busy with my job and my music and therefore have to cancel my plans on working on LMMS. Wish you the best for the future.

**Comment by JohannesLorenz** [comment]
@itlives Np. If you have anything unfinished we could use, feel free to send us a link to your branch.

At all: this issue is available again.

**Comment by CYBERDEViLNL** [comment]
Ok I'll bite :-)

**Comment by CYBERDEViLNL** [comment]
I’ve made a branch here https://github.com/CYBERDEViLNL/lmms/tree/master_dev

Should I create a PR against the `lv2` branch?

**Comment by JohannesLorenz** [comment]
Great :+1: 

Can you please make a PR against master? I'd say let's do Lv2 later.

Two things that I saw already:
* Lv2 instruments will also be loaded in `addNativePlugins`, which is why
  - I'd remove the "native" from the function name
  - it should have some code like around line 178 in the cpp file
```
for (const PluginKey& key : pluginKeys)
{
if(key.desc->name == "lv2instrument")
  // somehow add to m_lv2Root
else
  // somehow add to m_lmmsRoot
}
```
* `PluginDescList` got unused in your code. You can remove the class from the header.

**Comment by CYBERDEViLNL** [comment]
Good :-)

Thanks for the review, I’d made some changes and opened a PR here https://github.com/LMMS/lmms/pull/4923

**Comment by JohannesLorenz** [comment]
Many thanks @CYBERDEViLNL :rocket:

Can you please make a 2nd PR from your branch to branch `lv2-review` on remote `https://github.com/JohannesLorenz/lmms.git`?

**Comment by CYBERDEViLNL** [comment]
OK, NP glad I could help :-)

Done :+1:

**Comment by JohannesLorenz** [comment]
Just wanted to post an image how it looks like :smile: 

![Tree layout and search bar in action](https://i.imgur.com/ePU6DvQ.png)

**Comment by JohannesLorenz** [comment]
Already closed in April (!) in #4923 . Closing :smile:
